packages/node/src/ethereum/api.ethereum.ts

```diff
@@ -265,13 +265,8 @@ export class EthereumApi implements ApiWrapper<EthereumBlockWrapper> {
     includeTx?: boolean,
   ): Promise<EthereumBlockWrapped> {
     try {
-      const [block, logs] = await Promise.all([
-        this.getBlockPromise(blockNumber, includeTx),
-        this.client.getLogs({
-          fromBlock: hexValue(blockNumber),
-          toBlock: hexValue(blockNumber),
-        }),
-      ]);
+      const block = await this.getBlockPromise(blockNumber, includeTx);
+      const logs = await this.client.getLogs({ blockHash: block.hash });
```

Comment:
`infura / alchemy` doesn't seem support this parameter.
also, why remove Promise.all?

Reply:
We need to get the block hash to make the request

Reply:
no problem from me then.